homeassistant/core.py

```diff
@@ -190,6 +190,8 @@ def _async_add_job(self, target: Callable[..., None], *args: Any) -> None:
             self.loop.create_task(target)
         elif is_callback(target):
             self.loop.call_soon(target, *args)
+        elif isinstance(target, asyncio.Future):
```

Comment:
We should not add this here, as it is fixing a symptom instead of solving the problem. Instead make sure that no code passes a future to `async_add_job`.